pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -284,33 +284,20 @@ func (l *ListOptionIndexer) ListByOptions(ctx context.Context, lo ListOptions, p
 	}
 
 	// 2- Sorting: ORDER BY clauses (from lo.Sort)
-	orderByClauses := []string{}
-	if len(lo.Sort.PrimaryField) > 0 {
-		columnName := toColumnName(lo.Sort.PrimaryField)
-		if err := l.validateColumn(columnName); err != nil {
-			return nil, 0, "", err
-		}
-
-		direction := "ASC"
-		if lo.Sort.PrimaryOrder == DESC {
-			direction = "DESC"
-		}
-		orderByClauses = append(orderByClauses, fmt.Sprintf(`f."%s" %s`, columnName, direction))
-	}
-	if len(lo.Sort.SecondaryField) > 0 {
-		columnName := toColumnName(lo.Sort.SecondaryField)
-		if err := l.validateColumn(columnName); err != nil {
-			return nil, 0, "", err
-		}
+	if len(lo.Sort.Fields) > 0 {
+		orderByClauses := []string{}
+		for i, field := range lo.Sort.Fields {
+			columnName := toColumnName(field)
+			if err := l.validateColumn(columnName); err != nil {
+				return nil, 0, "", err
+			}
 
-		direction := "ASC"
-		if lo.Sort.SecondaryOrder == DESC {
-			direction = "DESC"
+			direction := "ASC"
+			if lo.Sort.Orders[i] == DESC {
```

Comment:
We should check that len(lo.Sort.Fields) == len(lo.Sort.Orders) and return an error, otherwise we (although unlikely) risk panicking.

Reply:
Done